### PR TITLE
improves LM input validation

### LINKED ISSUE #1918: Validation Message Needed If URL for LM is Too Long
Users have been pasting in long generated URL's that have exceeded the maximum length we can store.

We need to provide a message indicating that this is a fact and advising them to use tinyURL or something to shorten the length of the URL.

### PR DESCRIPTION
**Pull request by stopfstedt**
image

fixes #1918

### CODE REVIEW

**Review by dartajax — CHANGES_REQUESTED** [review]
If a 245 char URL is entered, it needs to wrap ...

image

**Comment by stopfstedt** [comment]
> If a 245 char URL is entered, it needs to wrap ...

different issue, please file separate ticket.

**Review by dartajax — APPROVED** [review]
if the main issue being fixed here is the 256 char max on URL, I approve

**Comment by stopfstedt** [comment]
> if the main issue being fixed here is the 256 char max on URL, I approve

new issue: https://github.com/ilios/common/issues/1923

File: tests/integration/components/new-learningmaterial-test.js [hunk 1]
```diff
@@ -39,4 +42,29 @@ module('Integration | Component | new learningmaterial', function (hooks) {
...
+        @cancel={{noop}}
+      />
+   `);
+    assert.equal(component.url.validationErrors.length, 0);
+    await component.save();
+    assert.equal(component.url.validationErrors.length, 1);
+    assert.equal(component.url.validationErrors[0].text, 'This field must be a valid url');
+    await component.url.set('https://validurl.edu/');
+    assert.equal(component.url.validationErrors.length, 0);
+    await component.url.set('https://validurl.edu/but-way-too-long/' + '0123456789'.repeat(25));
```
Comment (jrjohnson) [inline tests/integration/components/new-learningmaterial-test.js:63]: 💚

**Pull request merged**